### ISSUE TITLE
How to set UserAgent in flask-mwoauth

### ISSUE THREAD

**Issue by Jayprakash-SE** [valhallasw/flask-mwoauth#24]
Hey, I am getting a warning.
```
Sending requests with default User-Agent.  Set 'user_agent' on mwoauth.flask.MWOAuth to quiet this message.
```
So How to set user-agent to rid this message

**Comment by valhallasw** [comment]
Hi Jayprakash-SE,

Sorry for the delayed response. It seems the warning originates from `https://github.com/mediawiki-utilities/python-mwoauth/blob/6ba3cb8af0cb3b4b46572db2ff59edba605256eb/mwoauth/handshaker.py#L58`, which this library uses as underlying library. I think it requires a change to flask-mwoauth, specifically, adding the user-agent in the call to `https://github.com/valhallasw/flask-mwoauth/blob/master/flask_mwoauth/__init__.py#L40`

At the same time, it seems the `python-mwoauth` library now has support for flask built-in, so it may be a better option to use it directly. @halfak, can you chime in on this?

**Comment by halfak** [comment]
Hey folks!  Indeed, we made "user_agent" raise a warning for mwoauth given that it's  a soft requirement for Wikimedia projects (like Wikipedia), but if you're working with your own MediaWiki install, you can safely ignore the warning.  

If you did want to use mwoauth's flask blueprint directly, see https://pythonhosted.org/mwoauth/#module-mwoauth.flask  It provides a simple set of paths for working with mwoauth in a flash app.

**Comment by valhallasw** [comment]
@halfak: Thanks for the clarification! In your view, is there anything this (i.e., flask-mwoauth) flask blueprint offers above the one embedded in the main mwoauth one? Because if not, I think it makes more sense to redirect developers to the mwoauth one instead (which saves me the effort of maintaining ;-))

**Comment by halfak** [comment]
The goal of that module is to implement same basic functionality of flask-mwoauth.  I honestly can't remember *why* I decided to re-implement that functionality.  I guess I had originally intended to replace https://github.com/wikimedia/wikilabels/blob/master/wikilabels/wsgi/routes/auth.py with the blueprint and for some reason flask-mwoauth wasn't going to work for that.

Either way, I do intend to maintain this module, so I'd be happy (if that is the right word) to take on the responsibility.

**Comment by valhallasw** [comment]
I have added a comment suggesting to use python-mwoauth directly.

With regards to this specific ticket -- I'd be happy to merge a pull request, but I unfortunately do not have the time at the moment to write and test the code myself.

**Comment by urbanecm** [comment]
Fixed with cf2298f4cf55f4042b410d1ced8954a368b306e6.
